Last-Update: 2015-03-29
Forwarded: not-needed
Origin: upstream
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: fix FTBFS due to missing library "gnutls-openssl".
 ~~~~
 /usr/lib/ccache/c++   -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2  -Wall -D__STDC_CONSTANT_MACROS -O2   -fPIE -pie -Wl,-z,relro -Wl,-z,now -Wl,--as-needed CMakeFiles/nph-zms.dir/zms.cpp.o  -o nph-zms -rdynamic libzm.a -lz -lcurl -ljpeg -lpthread -lpcre -lgcrypt -lgnutls -lmysqlclient -lavformat -lavcodec -lavdevice -lavutil -lswscale
 libzm.a(zm_rtsp_auth.cpp.o): In function `Authenticator::computeDigestResponse(std::string&, std::string&)':
 /tmp/buildd/zoneminder-1.28.1/src/zm_rtsp_auth.cpp:149: undefined reference to `MD5'
 /tmp/buildd/zoneminder-1.28.1/src/zm_rtsp_auth.cpp:165: undefined reference to `MD5'
 /tmp/buildd/zoneminder-1.28.1/src/zm_rtsp_auth.cpp:187: undefined reference to `MD5'
 libzm.a(zm_user.cpp.o): In function `zmLoadAuthUser(char const*, bool)':
 /tmp/buildd/zoneminder-1.28.1/src/zm_user.cpp:225: undefined reference to `MD5'
 collect2: error: ld returned 1 exit status
 src/CMakeFiles/nph-zms.dir/build.make:102: recipe for target 'src/nph-zms' failed
 ~~~~

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,9 +224,13 @@
 	set(optlibsnotfound "${optlibsnotfound} GCrypt")
 endif(GCRYPT_LIBRARIES)
 
 # gnutls (using find_library and find_path)
-find_library(GNUTLS_LIBRARIES gnutls)
+find_library(GNUTLS_LIBRARIES gnutls-openssl)
+if(NOT GNUTLS_LIBRARIES)
+    find_library(GNUTLS_LIBRARIES gnutls)
+endif(NOT GNUTLS_LIBRARIES)
+
 if(GNUTLS_LIBRARIES)
 	set(HAVE_LIBGNUTLS 1)
 	list(APPEND ZM_BIN_LIBS "${GNUTLS_LIBRARIES}")
 	find_path(GNUTLS_INCLUDE_DIR gnutls/gnutls.h)
